qiskit/ignis/verification/topological_codes/fitters.py

```diff
@@ -159,46 +159,91 @@ def _make_syndrome_graph(self):
 
         return S
 
-    def weight_syndrome_graph(self, results):
-        """Generate weighted syndrome graph from result counts.
+    def get_error_probs(self, results):
+        """
+        Generate probabilities of single error events from result counts.
 
         Args:
             results (dict): A results dictionary, as produced by the
             `process_results` method of the code.
 
+        Returns:
+            dict: Keys are the edges for specific error
+            events, and values are the calculated probabilities
+
         Additional information:
             Uses `results` to estimate the probability of the errors that
-            create the pairs of nodes in S. The edge weights are then
-            replaced with the corresponding -log(p/(1-p).
+            create the pairs of nodes specified by the edge.
+            Calculation done using the method of Spitz, et al.
+            https://doi.org/10.1002/qute.201800012
         """
 
         results = results['0']
+        shots = sum(results.values())
+
+        nodes = self.S.nodes()
 
-        count = {element: {edge: 0 for edge in self.S.edges}
-                 for element in ['00', '01', '10', '11']}
+        error_probs = {}
+        for edge in self.S.edge_list():
 
-        for string in results:
+            # initialize averages
+            av_vv = 0  # v_ij
+            av_v = [0, 0]  # [v_,v_j]
+            av_xor = 0  # v_{i xor j}
 
-            nodes = self._string2nodes(string)
+            for string in results:
 
-            for edge in self.S.edge_list():
-                element = ''
-                for j in range(2):
-                    if edge[j] in nodes:
-                        element += '1'
-                    else:
-                        element += '0'
-                count[element][edge] += results[string]
+                # list of i for which v_i=1
+                error_nodes = self._string2nodes(string)
+
+                # get [v_i,v_j] for edge (i,j)
+                v = [int(nodes[edge[k]] in error_nodes) for k in range(2)]
```

Comment:
So one thing that is a potential source of bugs here is that there isn't a guarantee in retworkx that the node ids are actually contiguous. In case of node removals there can be holes in the node indices which is why a retworkx graph operates like a mapping instead of a sequence. I would probably delete L184 and just use:

```suggestion
                v = [int(self.S[edge[k]] in error_nodes) for k in range(2)]
```